src/hooks/v1/contractReader/UserHasPermission.ts

```diff
@@ -0,0 +1,32 @@
+import { NetworkContext } from 'contexts/networkContext'
+import { useContext } from 'react'
+import { V1ProjectContext } from 'contexts/v1/projectContext'
+import { V1OperatorPermission } from 'models/v1/permissions'
+
+import { useProjectOwner } from './ProjectOwner'
+import { useHasPermissions } from './HasPermissions'
+
+export function useUserHasPermission(
+  permission: V1OperatorPermission | V1OperatorPermission[],
+) {
+  const { userAddress } = useContext(NetworkContext)
+  const { projectId, isPreviewMode } = useContext(V1ProjectContext)
+
+  const { owner } = useProjectOwner()
+
+  const hasOperatorPermission = useHasPermissions({
+    operator: userAddress,
+    account: owner,
+    domain: projectId,
+    permissionIndexes: Array.isArray(permission) ? permission : [permission],
+  })
```

Comment:
This file was renamed from `HasPermission` to `UserHasPermission`

This is the only meaningful code change. We use the new `useHasPermissions` hook internally.